lib/elixir/lib/module/types/apply.ex

```diff
@@ -566,11 +566,21 @@ defmodule Module.Types.Apply do
     is_map_key(builtin_modules(), module)
   end
 
+  @builtin_protocols [
+    Collectable,
+    Enumerable,
+    IEx.Info,
+    Inspect,
+    JSON.Encoder,
+    List.Chars,
+    String.Chars
+  ]
```

Comment:
Is this gonna be nasty to keep up to date? I don't think so because I don't think we define new protocols often enough (2 out of those in the past ~10 years are new I think ahah), but just wanted to call it out.

Reply:
I asked myself the same question but there isn't also a cheap way for us to get all protocols. :(

Reply:
What about a test then?

```elixir
  test "protocols" do
    builtin_protocols =
      for app <- ~w[eex elixir ex_unit iex logger mix]a,
          Application.ensure_loaded(app),
          module <- Application.spec(app, :modules),
          Code.ensure_loaded(module),
          function_exported?(module, :__protocol__, 1),
          do: module

    # NOTE: if this test fails, update this list and lib/elixir/lib/module/types/apply.ex.
    assert Enum.sort(builtin_protocols) == [
             Collectable,
             Enumerable,
             IEx.Info,
             Inspect,
             JSON.Encoder,
             List.Chars,
             String.Chars
           ]
  end
```

Reply:
Great idea, test added!